File: GanAndroidApp/app/src/main/java/com/xattacker/binary/BinaryWritable.java
package com.xattacker.binary;

import java.io.IOException;
import java.io.UnsupportedEncodingException;

public interface BinaryWritable
{
   void writeBinary(byte[] aData, int aOffset, int aLength) throws IOException;
	void writeShort(short aShort) throws IOException;
	void writeInteger(int aInt) throws IOException;
	void writeLong(long aLong) throws IOException;
	void writeDouble(double aDouble) throws IOException;
	void writeString(String aStr) throws IOException, UnsupportedEncodingException;
}

File: GanAndroidApp/app/src/main/java/com/xattacker/binary/BinaryWritable.kt
package com.xattacker.binary

import java.io.IOException

// fluent interface
interface BinaryWritable<T>
{
    @Throws(IOException::class)
    fun writeBinary(aData: ByteArray, aOffset: Int, aLength: Int): T

    @Throws(IOException::class)
    fun writeShort(aShort: Short): T

    @Throws(IOException::class)
    fun writeInteger(aInt: Int): T

    @Throws(IOException::class)
    fun writeLong(aLong: Long): T

    @Throws(IOException::class)
    fun writeDouble(aDouble: Double): T

    @Throws(IOException::class)
    fun writeString(aStr: String): T
}
